lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -1364,6 +1362,7 @@ impl<'s> SegmentHolder {
             } else {
                 last_point_id_opt = Some(point_id);
                 last_segment_id_opt = Some(segment_id);
+                last_point_version_opt = None;
```

Comment:
The fix: properly clear point version when moving to the next point ID.

Thanks @generall for catching this one!